### ISSUE TITLE
Metadata not returned in case of non-ok status

### ISSUE THREAD

**Issue by janchochol** [tsloughter/grpcbox#30]
Currently code (e.g. in `grpcbox_client:unary_handler/6`) contains code:
```erlang
case recv_trailers(S) of
    {ok, {<<"0">>, _, Metadata}} ->
        {ok, Data} = recv_data(S, 0),
        {ok, Data, #{headers => Headers,
                     trailers => Metadata}};
    {ok, {Status, Message, _Metadata}} ->
        {error, {Status, Message}}
end;
```
Which means, that if status is not `?GRPC_STATUS_OK`, it does not send metadata (headers and trailers) to client.
We are integrating Erlang service with Kotlin (Java) services, which use additional trailers to send more details about errors.
With this code, we can not consume this information.
It can be fixed, e.g. by:
```erlang
case recv_trailers(S) of
    {ok, {<<"0">>, _, Metadata}} ->
        {ok, Data} = recv_data(S, 0),
        {ok, Data, #{headers => Headers,
                     trailers => Metadata}};
    {ok, {Status, Message, Metadata}} ->
        {error, {Status, Message}, #{headers => Headers, trailers => Metadata}}
end;
```
But unfortunately, it changes API of `grpcbox`.

We are willing to make pull request for this change, but we are not sure, how to approach API change.
There are at least two options:
* return always extended error, and handle API change through change in version
* use error format based on `Options` (add new option for extended error)
* maybe something else

What is your preferred approach?

**Comment by tsloughter** [comment]
A PR would be great. Breaking the API is fine, it isn't 1.0 and I've been working on a client "rewrite" because its the least thought out and well done of the 2 (server/client).

**Comment by janchochol** [comment]
Thanks, #31 created.

**Comment by janchochol** [comment]
Fixed by #31 .
